dask/array/percentile.py

```diff
@@ -207,14 +209,16 @@ def merge_percentiles(finalq, qs, vals, interpolation="lower", Ns=None):
 
     L = list(zip(*[(q, val, N) for q, val, N in zip(qs, vals, Ns) if N]))
     if not L:
-        raise ValueError("No non-trivial arrays found")
+        if raise_on_nan:
+            raise ValueError("No non-trivial arrays found")
+        return np.full((len(qs[0]) - 2,), np.nan)
```

Comment:
Mild legibility nit, feel free to ignore if you disagree:

```suggestion
        return np.full(len(qs[0]) - 2, np.nan)
```

Reply:
nah that is good.